src/tribler/core/start_core.py

```diff
@@ -133,6 +134,9 @@ def run_tribler_core_session(api_port: int, api_key: str, state_dir: Path, gui_t
                 f'API key: "{api_key}". State dir: "{state_dir}". '
                 f'Core test mode: "{gui_test_mode}"')
 
+    slow_coro_detection.patch_asyncio()  # Track the current coroutine handled by asyncio
```

Comment:
Since we have environment variables, we can do a conditional call to patch asyncio and to start watching thread. Same for setting loop.set_debug(True)